cpp/src/arrow/util/formatting_util_test.cc

```diff
@@ -427,4 +427,42 @@ TEST(Formatting, Timestamp) {
   }
 }
 
+TEST(Formatting, Interval) {
+  using DayMilliseconds = DayTimeIntervalType::DayMilliseconds;
+  using MonthDayNanos = MonthDayNanoIntervalType::MonthDayNanos;
+
+  const int32_t max_int32 = std::numeric_limits<int32_t>::max();
+  const int32_t min_int32 = std::numeric_limits<int32_t>::min();
+  const int64_t max_int64 = std::numeric_limits<int64_t>::max();
+  const int64_t min_int64 = std::numeric_limits<int64_t>::min();
+  {
+    StringFormatter<MonthIntervalType> formatter(month_interval());
+
+    AssertFormatting(formatter, 0, "0m");
```

Comment:
Is it a recommended format for months? Should we use "M" instead?

Reply:
Actually, I suppose that doesn't make any sense. M seems preferable to m though so I'll do that.